test/test-setup.tsx

```diff
@@ -3,43 +3,54 @@
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
-import React, { Reducer, useMemo } from 'react';
-
+import { renderHook } from '@testing-library/react-hooks';
+import React, { useMemo } from 'react';
 import { render, RenderResult } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { ModalManager, ThemeProvider, SnackbarManager } from '@zextras/carbonio-design-system';
 import { I18nextProvider } from 'react-i18next';
 import { Provider } from 'react-redux';
+import { MemoryRouter, Route } from 'react-router-dom';
+import { Store } from 'redux';
+import { useOnClickNewButton } from '../../hooks/on-click-new-button';
 import I18nTestFactory from './i18n/i18n-test-factory';
 
 interface ProvidersWrapperProps {
 	children?: React.ReactElement;
+	options?: any;
 }
 
-function customRender(ui: React.ReactElement, options?: any): RenderResult {
-	const ProvidersWrapper = ({ children }: ProvidersWrapperProps): JSX.Element => {
-		const i18n = useMemo(() => {
-			const i18nFactory = new I18nTestFactory();
-			return i18nFactory.getAppI18n();
-		}, []);
-
-		const store = options?.store ?? {};
-
-		return (
-			<ThemeProvider>
-				<Provider store={store}>
-					<I18nextProvider i18n={i18n}>
-						<SnackbarManager>
-							<ModalManager>{children}</ModalManager>
-						</SnackbarManager>
-					</I18nextProvider>
-				</Provider>
-			</ThemeProvider>
-		);
-	};
+export const ProvidersWrapper = ({ children, options }: ProvidersWrapperProps): JSX.Element => {
+	const { store = {}, initialEntries = ['/calendars'], path = '/calendars' } = options;
```

Comment:
Does this make sense only for Calendars? Could we design a way for the modules to customize the path?